### PR TITLE
Editable flow run names

### PR DESCRIPTION
**Pull request by znicholasbrown**
PR Checklist:

- [x] add a short description of what's changed to the top of the `CHANGELOG.md`
- [ ] add/update tests (or don't, for reasons explained below)

## Describe this PR
Enables modifying flow run names directly from the flow run page:
Nov-17-2020 13-14-15

Also introduces a new "editable field" component, which is a drop-in alternative to `v-input` meant to maintain inline styling while allowing modification _without_ using `contenteditable` (which doesn't work with many Vue-related APIs)

### CODE REVIEW

**Review by cicdw — APPROVED** [review]
Others may have comments on the code itself, but from a functionality perspective this is awesome ✅

**Review by zhen0 — CHANGES_REQUESTED** [review]
The flow run name editing looks great @znicholasbrown. One requested change for the jsonInput component (your changes would currently break the highlight on focus.)

File: src/components/JsonInput.vue [hunk 1]
```diff
@@ -179,15 +179,15 @@ export default {
...
-        'blue-border': prependIcon && focussed && !jsonError,
+        'blue-border': prependIcon && focused && !jsonError,
         'red-border': prependIcon && jsonError,
-        'plain-border': prependIcon && !focussed && !jsonError,
+        'plain-border': prependIcon && !focused && !jsonError,
         'original-border': !prependIcon
       }"
       :options="editorOptions"
       @input="handleJsonInput($event)"
-      @focus="focussed = true"
```
[the remark this answers is not in the document]
Comment (znicholasbrown) [reply src/components/JsonInput.vue:189]: TIL that's not a typo 😆  - will revert that!

**Review by zhen0 — APPROVED** [review]
LGTM @znicholasbrown  - just need to update master

**Pull request merged**